### PR TITLE
validation and exception handler added

### PR DESCRIPTION
**Pull request by maiiat**

### CODE REVIEW

File: src/main/java/com/example/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
```
Comment (kerrrusha) [inline src/main/java/com/example/dto/CreateBookRequestDto.java:15]: not blank is more preferred as it checks string for not being null, not empty and not have only spaces
```suggestion
    @NotBlank
    private String title;
    @NotBlank
    private String author;
    @NotBlank
```

File: src/main/java/com/example/dto/UpdateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class UpdateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
```
Comment (kerrrusha) [inline src/main/java/com/example/dto/UpdateBookRequestDto.java:15]: same

File: src/main/java/com/example/dto/UpdateBookRequestDto.java [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+    @Isbn
     private String isbn;
     private String description;
     private String coverImage;
+    @NotNull
+    @Min(0)
```
Comment (kerrrusha) [inline src/main/java/com/example/dto/UpdateBookRequestDto.java:21]: same

File: src/main/java/com/example/dto/UpdateBookRequestDto.java
Comment (kerrrusha) [inline src/main/java/com/example/dto/UpdateBookRequestDto.java:1]: this have identic fields as in create book dto, maybe we can combine them

File: src/main/java/com/example/validation/IsbnValidator.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public void initialize(Isbn constraintAnnotation) {
+        // initialization logic if needed
+    }
+
+    @Override
+    public boolean isValid(String isbn, ConstraintValidatorContext context) {
+        if (isbn == null) {
+            return true; // @NotNull should handle null case
```
Comment (kerrrusha) [inline src/main/java/com/example/validation/IsbnValidator.java:27]: please remove comments

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/validation/IsbnValidator.java (hunk 4 above)
Comment (maiiat) [reply src/main/java/com/example/validation/IsbnValidator.java:27]: done

File: src/main/java/com/example/dto/UpdateBookRequestDto.java
Comment (maiiat) [reply src/main/java/com/example/dto/UpdateBookRequestDto.java:1]: done

File: src/main/java/com/example/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (maiiat) [reply src/main/java/com/example/dto/CreateBookRequestDto.java:15]: done

File: src/main/java/com/example/dto/UpdateBookRequestDto.java (hunk 2 above)
Comment (maiiat) [reply src/main/java/com/example/dto/UpdateBookRequestDto.java:15]: done

File: src/main/java/com/example/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,14 +1,23 @@
...
+    @Isbn
     private String isbn;
     private String description;
     private String coverImage;
+    @NotNull
+    @Min(0)
```
[the remark this answers is not in the document]
Comment (maiiat) [reply src/main/java/com/example/dto/CreateBookRequestDto.java:21]: done

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**